app/routes/surveys/components/Submissions/SubmissionPublicResults.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import * as React from 'react';
+import styles from '../surveys.css';
+import type { SurveyEntity } from 'app/reducers/surveys';
+import { Content, ContentSection, ContentMain } from 'app/components/Content';
+import type { ActionGrant } from 'app/models';
+import Results from './Results';
+import { QuestionTypes, TokenNavigation } from '../../utils';
+
+type Props = {
+  survey: SurveyEntity,
+  actionGrant: ActionGrant,
+  results: Object
+};
+
+const SubmissionPublicResultsPage = ({ survey, actionGrant }: Props) => {
+  const results = survey.results || {};
+
+  const generateTextAnswers = question => {
+    let texts = [];
+    Object.keys(results[String(question.id)])
+      .map(name => {
+        if (name !== 'questionType') {
+          texts = results[String(question.id)][name].map((answer, i) => (
+            <li key={i}>{answer}</li>
+          ));
+        }
+      })
+      .filter(Boolean);
+
+    return texts.length === 0 ? <i>Ingen svar.</i> : texts;
+  };
+
+  const generateQuestionData = questionId => {
```

Comment:
Would be nice if all this logic was in a separate file. The components are quite large, and hard to read from A to B in order to understand what component does what.

Reply:
You sure? In terms of what the purpose of each file is, I think the structure is pretty nice. I understand readability concerns when reviewing, but is it really any easier to get a good sense of what's going on when functionality very clearly tied to the purpose of a file is moved out to a `utils` file or something where they have to take in a boatload of parameters that would otherwise be available as props?